icu4c/source/common/dictbe.cpp

```diff
@@ -1053,13 +1064,18 @@ CjkBreakEngine::CjkBreakEngine(DictionaryMatcher *adoptDictionary, LanguageType
     // Korean dictionary only includes Hangul syllables
     fHangulWordSet.applyPattern(UnicodeString(u"[\\uac00-\\ud7a3]"), status);
     fHangulWordSet.compact();
+    fHiraganaWordSet.applyPattern(UnicodeString(u"[:Hiragana:]"), status);
+    fHiraganaWordSet.compact();
+    fNumberWordSet.applyPattern(UnicodeString(u"[:Nd:]"), status);
+    fNumberWordSet.compact();
 
     // handle Korean and Japanese/Chinese using different dictionaries
     if (type == kKorean) {
         if (U_SUCCESS(status)) {
             setCharacters(fHangulWordSet);
         }
     } else { //Chinese and Japanese
+        loadJpParticleAndAuxVerbs();
```

Comment:
Move this line into the if (U_SUCCESS(status)) { block and pass the status by reference to that function.